core/src/commonMain/kotlin/com/xebia/functional/xef/auto/CoreAIScope.kt

```diff
@@ -159,4 +161,22 @@ constructor(
     size: String = "1024x1024",
     promptConfiguration: PromptConfiguration = PromptConfiguration.DEFAULTS
   ): ImagesGenerationResponse = images(prompt, context, numberImages, size, promptConfiguration)
+
+  private val finalizers: Atomic<List<() -> Unit>> = Atomic(emptyList())
```

Comment:
I added the implementation here, but I don't know if it would make sense to have this implementation somewhere else. In that case, where should we have that? I guess that it should be in the `xef-core` module, but which package?

Reply:
I would add it in a seperate file/class and use Kotlin delegation. So I would just copy the gist I shared 1-on-1 and then you can enable the syntax in this class using delegation.

```kotlin
: AutoCloseable, AutoClose by AutoClose() {
```

Well, almost 1-on-1. For the above syntax to work you'd need `AutoClose` to be an `interface` instead of a class and then have a `fun AutoClose(): AutoClose` smart-constructor which provides the implementation for the interface. I updated the gist, https://gist.github.com/nomisRev/952c619fa40c7c0041974992a6ad1e06.